dspace-api/src/main/java/org/dspace/util/SolrUpgradeStatistics6.java

```diff
@@ -0,0 +1,610 @@
+                        while(run(Constants.COMMUNITY) > 0){};
+                }
+
+                refreshContext();
+                
+                if (numProcessed > 0) {
+                        printTime(String.format("\n\t* Total Processed: %,12d", numProcessed), true);
+                        runReport();
+                }
+        }
+
+        /*
+         * Report on the existence of legacy id records within a shard
+         */
+        private void runReport() throws SolrServerException {
+                System.out.println();
+                System.out.println("=================================================================");
+                System.out.println("\t*** Statistics Records with Legacy Id ***\n");
+                int total = runReport(false);
+                total += runReport(true);
+                System.out.println("\t--------------------------------------");
+                System.out.println(String.format("\t%,12d\t%s", total, "TOTAL"));
+                System.out.println("=================================================================");
+                System.out.println();
+        }
+        /*
+         * Report on the existence of specific legacy id records within a shard
+         * @param search If true, report on search scope records.  Otherwise report on view (browse) records
+         */
+        private int runReport(boolean search) throws SolrServerException {
+                String field = search ? "scopeId" : "id";
+                String query = String.format("NOT(%s:*-*)", field); 
+                SolrQuery sQ = new SolrQuery();
+                sQ.setQuery(query);
+                sQ.setFacet(true);
+                sQ.addFacetField(search? "scopeType" : "type");
+                QueryResponse sr = server.query(sQ);
+                
+                int total = 0;
+                for(FacetField ff: sr.getFacetFields()) {
+                        for(FacetField.Count count: ff.getValues()) {
+                                String name = count.getName();
+                                int id = Integer.parseInt(name);
+                                String s = search ? "Search" : "View";
+                                if (id == Constants.COMMUNITY) {
+                                        name = "Community " + s;
+                                } else if (id == Constants.COLLECTION) {
+                                        name = "Collection " + s;
+                                } else if (id == Constants.ITEM) {
+                                        name = "Item " + s;
+                                } else if (id == Constants.BITSTREAM) {
+                                        name = "Bistream " + s;
+                                } else if (id == Constants.SITE) {
+                                        //I have inconsistent values for this field
+                                        name = "Site "+ s + "(TBD)";
```

Comment:
I have not yet integrated the 5.7 fixes to solr export, so I am having difficulty porting some odd conditions into my test environment.  Once I get the other issues resolved, I will try to duplicate this condition into my test environment.